File: go/bfs/min_dist_test.go
package bfs

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestMinDist_RandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData()

		expected := td.minDist
		actual := MinDist(td.graph, td.start, td.end)

		if expected != actual {
			assert.FailNowf(t, "",
				"MinDist(%v, %v, %v):\nwant %v\n got %v",
				td.graph, td.start, td.end, expected, actual,
			)
		}
	}
}

File: python/bfs/min_dist_test.py
from min_dist import min_dist
from random_test_data import RandomTestData


def test_min_dist():
	for _ in range(10_000):
		td = RandomTestData()

		want = td.min_dist
		got = min_dist(td.graph, td.start, td.end)

		if got != want:
			assert False, f'minDist({td.graph}, {td.start}, {td.end}):\nwant {want}\n got {got}'
